cmd/vc-rest/startcmd/start.go

```diff
@@ -338,9 +348,52 @@ func buildEchoHandler(conf *Configuration, cmd *cobra.Command) (*echo.Echo, erro
 		devapi.RegisterHandlers(e, devController)
 	}
 
+	metricsProvider, err := NewMetricsProvider(conf.StartupParameters, defaultVCSKeyManager.KeyManager)
+	if err != nil {
+		return nil, err
+	}
+
+	if metricsProvider != nil {
+		err = metricsProvider.Create()
+		if err != nil {
+			return nil, err
+		}
+	}
+
 	return e, nil
 }
 
+type keyManager interface {
+	Create(kt kms.KeyType, opts ...kms.KeyOpts) (string, interface{}, error)
+	Get(keyID string) (interface{}, error)
+	ExportPubKeyBytes(keyID string) ([]byte, kms.KeyType, error)
+	ImportPrivateKey(privKey interface{}, kt kms.KeyType, opts ...kms.PrivateKeyOpts) (string, interface{}, error)
+	HealthCheck() error
+}
+
+func NewMetrics(parameters *startupParameters) (metricsProvider.Metrics, error) {
+	switch parameters.metricsProviderName {
+	case "prometheus":
+		return promMetricsProvider.GetMetrics(), nil
+	default:
+		return noopMetricsProvider.GetMetrics(), nil
+	}
+}
+
+func NewMetricsProvider(parameters *startupParameters, km keyManager) (metricsProvider.Provider, error) {
+	switch parameters.metricsProviderName {
+	case "prometheus":
+		metricsHttpServer := httpserver.New(
```

Comment:
add echo server here. No need for httpserver pkg

Reply:
updated